src/flowchem/components/valves/valve.py

```diff
@@ -219,8 +226,9 @@ async def get_position(self) -> tuple[tuple, tuple]:
         pos = int(pos) if pos.isnumeric() else pos
         return self._positions[int(self._change_connections(pos, reverse=True))]
 
-
-    async def set_position(self, connect: str | tuple = "", disconnect: str | tuple = "",
+    async def set_position(self,
```

Comment:
while this works, would it not be possible to get a better type description? So the thing is, the set_position function actually acts on nested lists, it is adapted (I think) to also accept the str representation of nested lists. This however is not evident. While a subclass of str is accepted, it is very incomplete to say it accepts str

Reply:
complex types are no accepted by fastapi and the endpoint is not created @samuelvitorsaraiva if you can describe better would be helpful

Reply:
FastAPI does not support nested lists in query parameters

The issue arises from how lists are passed via URL query strings in FastAPI.

For example, consider the following method:

async def set_position(self, connect, ...)

If we try to send an argument like `connect = [1, 0]` using the URL:

http://127.0.0.1:8000/items/?connect=1&connect=0

What actually gets passed into the method is `connect = 0`. (This behavior was tested.) FastAPI does not parse the list correctly in this case.

Additionally, if we declare the argument type as:

... connect: list[int]  or connect: list[list[int]] ...

The argument does not even appear available in the API documentation, and in some cases, it causes the API initialization to break entirely.

Similar issues have been discussed in forums

https://stackoverflow.com/questions/74759875/how-to-send-a-list-of-lists-as-query-parameter-in-fastapi